src/internal.c

```diff
@@ -5818,6 +5830,15 @@ int DhGenKeyPair(WOLFSSL* ssl, DhKey* dhKey,
 
     WOLFSSL_ENTER("DhGenKeyPair");
 
+#if defined(HAVE_PK_CALLBACKS)
+    if (ssl && ssl->ctx && ssl->ctx->DhGenerateKeyPairCb) {
+        ret = ssl->ctx->DhGenerateKeyPairCb(dhKey, ssl->rng, priv, privSz,
+                                            pub, pubSz);
+        if (ret != NOT_COMPILED_IN)
```

Comment:
What about when ret == NOT_COMPILED_IN? Do we want to fall back to wc_DhGenerateKeyPair? Because that's what'll happen as written.

Reply:
Yes.  I think so.  This gives hardware implementations a chance to validate that they can support the operation. For example, if the hardware does not support `ffdhe8192` then this gracefully allow fallback to wolfssl's implementation.